dlme_airflow/harvester/source_harvester.py

```diff
@@ -4,14 +4,18 @@
 PROVIDER = "provider"
 
 
-def data_source_harvester(**kwargs):
-    """Intake source harvester, takes a provider (for nested catalog use the catalog
-    name.source, i.e. bodleian.arabic), generates a Pandas DataFrame, runs
-    validations, and saves a copy of the DataFrame.
-
-    @param -- provider
+def data_source_harvester(task_instance, **kwargs):
+    """Intake source harvester, takes a Collection object, generates
+    a Pandas DataFrame from a harvest of it, and saves a copy of the
+    DataFrame.  Also, pushes some basic stats about the dataframe for
+    use in e.g. downstream validation of transform output.
     """
 
     collection = kwargs.get("collection")
-    # dataframe_to_file(source, collection.provider.name, collection)
-    return dataframe_to_file(collection)
+    df_and_csv = dataframe_to_file(collection)
+    df = df_and_csv["source_df"]
+    working_csv = df_and_csv["working_csv"]
```

Comment:
@edsu and @lwrubel also interested, if returning a hash is reasonably idiomatic, in whether there's a more idiomatic way to de-structure the returned hash.